cpp/kiss_icp/pipeline/KissICP.cpp

```diff
@@ -107,4 +107,11 @@ bool KissICP::HasMoved() {
     return motion > 5.0 * config_.min_motion_th;
 }
 
+void KissICP::Reset(Eigen::Quaterniond quaternion, Eigen::Vector3d translation) {
+    poses_.clear();
+    Sophus::SE3d predpred(quaternion, translation);
+    poses_.push_back(predpred);
+    local_map_.Clear();
```

Comment:
Instead of resetting the local map here, would in make more sense to transformed it the new reference frame so that KISS can continue his "job" with the old map points?

What do you think ? @nachovizzo @benemer @RamanaBotta

Reply:
I agree, we shold transform the local map and the poses to the new frame. Otherwise, every time `Reset` is called, the constant velocity model can not be applied and the map is empty. So you lose a lot of the nice properties of KISS.

Reply:
> Instead of resetting the local map here, would in make more sense to transformed it the new reference frame so that KISS can continue his "job" with the old map points?
> 
> What do you think ? @nachovizzo @benemer @RamanaBotta

I agree, It makes sense to transform the local map to the new reference frame.

Reply:
Hello @tizianoGuadagnino @benemer ,
I have tried transforming the local map to the new reference frame. Please look [here](https://github.com/RamanaBotta/kiss-icp/blob/6f17fcd32cd8c0c2059c386e7db61e3d194723a8/cpp/kiss_icp/pipeline/KissICP.cpp#L110) for implementation.
### Issue 
After resetting to the new frame, the Local map is not looking good, It has previous ref frame points too.
![Screenshot from 2023-10-14 00-13-14](https://github.com/PRBonn/kiss-icp/assets/78207982/67b11039-c09a-48c0-a3a2-dda8d85123f6)
### Help
Please consider looking into implementation [here](https://github.com/RamanaBotta/kiss-icp/blob/6f17fcd32cd8c0c2059c386e7db61e3d194723a8/cpp/kiss_icp/pipeline/KissICP.cpp#L110), please let me know any corrections or suggestions.

Reply:
Can you please push your changes to this branch? This makes it easier to discuss it.

For example, you are still clearing the poses when resetting the map. This will prevent the system from de-skewing and using a constant velocity model because the velocity can not be estimated without prior pose information. If it is desired to reset the list of estimated poses, I would at least transform and keep the last two.

Reply:
Hi @benemer, Can you Please reiterate this suggestion of transforming the local map, without this also things are working fine.

Reply:
Hi @RamanaBotta, for us it is not just important that it works but it also needs to be correct from the theoretical perspective. Once the pose has been reset, it just makes sense to transform the local map in the new reference frame, so that KISS can use the previous map. Please fix this.